Sources/ApolloWebSocket/WebSocketTransport.swift

```diff
@@ -291,13 +296,14 @@ extension WebSocketTransport: WebSocketDelegate {
   
   public func websocketDidConnect(socket: WebSocketClient) {
     self.error.value = nil
-    initServer()
+    initServer(reconnect: reconnect.value)
     if reconnected {
       self.delegate?.webSocketTransportDidReconnect(self)
       // re-send the subscriptions whenever we are re-connected
       // for the first connect, any subscriptions are already in queue
       for (_,msg) in self.subscriptions {
-        write(msg)
+        let id = allowSendingDuplicates ? nil : queue.first { $0.value == msg }?.key
```

Comment:
why send a `nil` id if duplicates are disallowed instead of just not sending at all?

Reply:
If Websocket reconnects then there is no entry in the queue so we need to call `write(msg, id: id)` which actually adds to queue and when `acked` then sends to the GraphQL Websockets. There is function `writeQueue()` that does it.

If it hasn't `acked` then queue isn't empty and we can check for duplicates. We need to call `write` in this case as well because while looping trough server can send `acked` back.

Does it make sense?

Reply:
Ah I think I flipped the ternary in my mind - I've always been shit with ternaries. 

So in this case if we *are* allowing the sending of duplicates, then we send a `nil` ID because it doesn't matter. If we're *not* allowing the sending of duplicates then we check through everything for a matching message to find its ID? 

It still strikes me as a bit weird to check the messages instead of the IDs, but that makes moderately more sense.

Reply:
Yes exactly what you wrote.

Ideally maybe we should link subscription IDs and IDs in the queue, but currently it  is not the case. That's why I figured out searching for message. As this could be opt-in approach I think for now it's fine.

What do you think? Are there any other folks who can help to review? Maybe I'm missing something.

I agree about ternary operator also not my cup of tea :P

Reply:
might be clearer if you got rid of the ternary and just explicitly sent `nil`

Reply:
Changed to simpler approach